sqlglot/expressions.py

```diff
@@ -1117,3 +1570,141 @@ def _all_functions():
 
 
 ALL_FUNCTIONS = _all_functions()
+
+
+def _maybe_parse(
+    code_or_expression,
+    *,
+    expression_type=None,
+    dialect=None,
+    prefix=None,
+    parser_opts=None,
+):
+    if isinstance(code_or_expression, Expression):
+        return code_or_expression
+
+    parser_opts = parser_opts or {}
+    code = str(code_or_expression)
+    if prefix:
+        code = f"{prefix} {code}"
+
+    from sqlglot.dialects import Dialect
+
+    dialect = Dialect.get_or_raise(dialect)()
+
+    if expression_type:
+        result = dialect.parse_into(expression_type, code, **parser_opts)
+    else:
+        result = dialect.parse(code, **parser_opts)
+
+    if not result:
+        return None
+    return result[0]
+
+
+T = TypeVar("T", bound=Expression)
+
+
+def _maybe_copy(instance: T, copy=True) -> T:
+    if copy:
+        instance = instance.transform(lambda node: node, copy=True)
```

Comment:
are you doing transform to recursively set parents? instead, can you just make sure that the copy() method recursively sets the parent? is that cleaner?